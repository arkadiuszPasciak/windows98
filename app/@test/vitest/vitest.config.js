import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
