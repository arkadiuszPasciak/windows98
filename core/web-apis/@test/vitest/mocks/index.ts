export * from "./canvas-api.mock"
export * from "./media-devices-api.mock"
export * from "./screen-api.mock"
export * from "./video-api.mock"
