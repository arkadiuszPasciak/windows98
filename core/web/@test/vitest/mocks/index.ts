export * from "./battery-status-api.mock"
export * from "./canvas-api.mock"
export * from "./clipboard-api.mock"
export * from "./cookie-store-api.mock"
export * from "./device-memory-api.mock"
export * from "./fullscreen-api.mock"
export * from "./location-api.mock"
export * from "./media-devices-api.mock"
export * from "./screen-api.mock"
export * from "./user-agent-client-hints-api.mock"
export * from "./video-api.mock"
export * from "./web-storage.mock"
