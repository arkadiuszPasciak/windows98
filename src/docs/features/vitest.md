# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
