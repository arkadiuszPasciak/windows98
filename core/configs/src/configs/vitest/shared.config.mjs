import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
