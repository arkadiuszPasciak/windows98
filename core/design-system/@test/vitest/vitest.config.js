import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"

export default mergeConfig(
	sharedConfig,
	defineProject({/* here custom configs */ })
)
