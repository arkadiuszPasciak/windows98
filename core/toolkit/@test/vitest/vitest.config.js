import sharedConfig from "@windows98/configs/vitest/shared"
import { defineProject, mergeConfig } from "vitest/config"

export default mergeConfig(
	sharedConfig,
	defineProject({
		/* here custom configs */
	}),
)
