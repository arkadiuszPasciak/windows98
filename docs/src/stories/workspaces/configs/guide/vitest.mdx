import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

