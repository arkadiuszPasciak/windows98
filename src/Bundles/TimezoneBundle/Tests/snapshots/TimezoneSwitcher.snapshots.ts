import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import Timezone from '@Bundles/TimezoneBundle/Components/TimezoneSwitcher.vue'

const defaultWrapper = mount(Timezone)

describe('[TimezoneBundle]<Snapshots>(TimezoneSwitcher)', async () => {
  it('renders correctly component', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
