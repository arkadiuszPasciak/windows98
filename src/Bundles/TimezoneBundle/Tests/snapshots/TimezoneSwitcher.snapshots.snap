// Vitest Snapshot v1

exports[`[TimezoneBundle]<Snapshots>(TimezoneSwitcher) > renders correctly component 1`] = `
<div
  class="UIFrame TimezoneSwitcher"
  data-v-8f4e71f6=""
  data-v-ff9b8841=""
>
  <p
    class="UIText title"
    data-test="frame-title-timezone"
    data-v-f0b878d5=""
    data-v-ff9b8841=""
  >
    
    Timezone
    
  </p>
  
  <div
    class="UISelect select"
    data-v-8f4e71f6=""
    data-v-d3c68ad1=""
  >
    <!--v-if-->
    <select
      class="select"
      data-test="ui-select-timezones"
      data-v-d3c68ad1=""
      id="timezones"
      name="timezones"
    >
      
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-12"
      >
        GMT-12 International Date Line West (IDLW)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-11"
      >
        GMT-11 Nome Time (NT)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-10"
      >
        GMT-10 Hawaii Standard Time (HST)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-9"
      >
        GMT-9 Alaska Standard Time (AKST)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-8"
      >
        GMT-8 Pacific Standard Time (PST)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-7"
      >
        GMT-7 Mountain Standard Time (MST)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-6"
      >
        GMT-6 Central Standard Time (CST)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-5"
      >
        GMT-5 Eastern Standard Time (EST)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-4"
      >
        GMT-4 Atlantic Standard Time (AST)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-3"
      >
        GMT-3 Argentina Time (ART)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-2"
      >
        GMT-2 Azores Time (AT)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="-1"
      >
        GMT-1 West Africa Time (WAT)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="0"
      >
        GMT+0 Greenwich Mean Time (GMT)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="1"
      >
        GMT+1 Central European Time (CET)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="2"
      >
        GMT+2 Eastern European Time (EET)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="3"
      >
        GMT+3 Moscow Time (MSK)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="4"
      >
        GMT+4 Armenia Time (AMT)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="5"
      >
        GMT+5 Pakistan Standard Time (PKT)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="6"
      >
        GMT+6 Omsk Time (OMSK)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="7"
      >
        GMT+7 Kranoyask Time (KRAT)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="8"
      >
        GMT+8 China Standard Time (CST)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="9"
      >
        GMT+9 Japan Standard Time (JST)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="10"
      >
        GMT+10 Eastern Australia Standard Time (AEST)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="11"
      >
        GMT+11 Sakhalin Time (SAKT)
      </option>
      <option
        class="option"
        data-v-d3c68ad1=""
        value="12"
      >
        GMT+12 New Zealand Standard Time (NZST)
      </option>
      
    </select>
  </div>
  
</div>
`;
